spec/models/edition_spec.rb

```diff
@@ -207,4 +207,40 @@
       end
     end
   end
+
+  describe "#assign_access_limit" do
+    it "can set an access limit" do
+      edition = build(:edition)
+      user = build(:user)
+
+      expect { edition.assign_access_limit(limit_type: :all_organisations, user: user) }
+        .to change { edition.access_limit }
+        .from(nil)
+        .to(an_instance_of(AccessLimit))
+
+      expect(edition.access_limit).to be_all_organisations
+      expect(edition.access_limit.created_by).to eq(user)
+      expect(edition.access_limit.revision_at_creation).to eq(edition.revision)
+    end
+
+    it "can remove an access limit" do
+      edition = build(:edition, :access_limited)
+
+      expect { edition.assign_access_limit(limit_type: nil, user: build(:user)) }
+        .to change { edition.access_limit }
+        .from(an_instance_of(AccessLimit))
+        .to(nil)
+    end
+
+    it "updates the edition last edited information" do
+      edition = build(:edition)
+      user = build(:user)
+
+      travel_to(Time.current) do
+        expect { edition.assign_access_limit(limit_type: :all_organisations, user: user) }
```

Comment:
Similarly here, could we separate the setup/run/test phases in here, so it's a bit clearer?

Reply:
I think this an appropriate way to test whether a method changes the state of an object. It has advantages that it checks something actually changes and the syntax is more declarative of what we're testing.